src/main/java/seedu/address/logic/commands/ClearAllCommand.java

```diff
@@ -0,0 +1,43 @@
+package seedu.address.logic.commands;
+
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+/**
+ * Clears every person in the address book.
+ */
+public class ClearAllCommand extends Command {
+
+    public static final String COMMAND_WORD = "clear ALL ENTRIES";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Clears all entries from the address book.\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_CLEAR_ALL_SUCCESS = "Successfully cleared all entries from the address book";
+
+    public ClearAllCommand() {
+
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (lastShownList.size() == 0) {
+            throw new CommandException(Messages.MESSAGE_EMPTY_LIST);
+        }
+
+        model.clearAllPerson();
+        return new CommandResult(MESSAGE_CLEAR_ALL_SUCCESS);
+    }
```

Comment:
Try to get the unfiltered list.
Using the filtered list may lead to this scenario:
1. There is data in the JSON file
2. User filters with `find` command and uses some nonsense keyword
3. User gets 0 search results back
4. lastShownList.size() == 0
5. Exception thrown saying the JSON file is empty when it is not empty